index-scheduler/src/batch.rs

```diff
@@ -1222,6 +1226,17 @@ impl IndexScheduler {
         for task in to_delete_tasks.iter() {
             self.all_tasks.delete(wtxn, &BEU32::new(task))?;
         }
+        for canceled_by in affected_canceled_by {
+            let canceled_by = BEU32::new(canceled_by);
+            if let Some(mut tasks) = self.canceled_by.get(wtxn, &canceled_by)? {
+                tasks -= &to_delete_tasks;
+                if tasks.is_empty() {
+                    self.canceled_by.delete(wtxn, &canceled_by)?;
+                } else {
+                    self.canceled_by.put(wtxn, &canceled_by, &tasks)?;
+                }
+            }
+        }
```

Comment:
Ok so here you clean up the deleted tasks from the lists of canceled tasks, right?